graphql_test.go

```diff
@@ -654,6 +654,37 @@ func TestCustomValidatingDirective(t *testing.T) {
 				{Message: `rebels cannot request imperial units`, Locations: []gqlerrors.Location{{Line: 58, Column: 3}}, Path: []interface{}{"hero", "height"}},
 			},
 		},
+		{
+			Schema: graphql.MustParseSchema(
+				`directive @gg_length(
+					max: Int
+					min: Int
+				) on ARGUMENT_DEFINITION
+
+				`+strings.ReplaceAll(
+					starwars.Schema,
+					"search(text: String!): [SearchResult]!",
+					`search(text: String! @gg_length(min: 3)): [SearchResult]!`,
+				),
+				&starwars.Resolver{},
+				graphql.Directives(&lengthDirective{}),
+			),
+			Query: `
+				query Search($text: String!) {
+					search(text: $text) {
+						... on Human { id name }
+						... on Droid { id name }
+						... on Starship { id name }
+					}
+				}
+			`,
+			Variables:      map[string]interface{}{"text": "Lu"},
+			ExpectedResult: "null",
+			ExpectedErrors: []*gqlerrors.QueryError{
+				// FIXME: would be helpful to include the name of the field? The context of the type / query around that also??
```

Comment:
The errors support extensions. You can include the name of the field and the name of the variable in the extensions. right?

Reply:
Actually, wouldn't the path include the arg name 🤔? I have to read the spec.

Reply:
We've got the `path` which I think is the _aliased_ path to the field, while I think I meant the name of the argument(s) here, which we don't know at all. We have the directive implementation, but not what it is attached to. `search` here takes `text`, but the directive implementation doesn't see that

Reply:
I agree with you on this one. Some solution will be needed for this problem.

Reply:
Latest commit should have sorted this out, as the function takes the ast type and can get the arg name from there